developer_guides/uuid/index.rst

```diff
@@ -0,0 +1,134 @@
+collisionn-free way of component identification. UUIDs are currently
+used as the standard interface by all users of the SOF firmware,
+including the tracing subsystem, the topology .m4 files, and the Linux
+topology driver.
+
+Allocating a new UUID for the new added component is mandatory, and
+UUIDs are expected to replace the component type in the near future. For
+the whole SOF stack, the usage of UUID should follow rules as below:
+
+UUID allocation
+***************
+The UUID allocation of a specific component should use the version 4 in
+https://en.wikipedia.org/wiki/Universally_unique_identifier, and the
+value should be specified in the firmware component dirver, e.g. for src
+component, in sof/src/audio/src.c:
+
+.. code-block:: none
+
+   /* c1c5326d-8390-46b4-aa47-95c3beca6550 */
+   DECLARE_SOF_UUID("src", src_uuid, 0xc1c5326d, 0x8390, 0x46b4,
+                    0xaa, 0x47, 0x95, 0xc3, 0xbe, 0xca, 0x65, 0x50);
+
+UUID in topology
+****************
+We should provide UUID string in topology .m4 file in the exactly same
+format as commented in the firmware, e.g for host it is
+c1c5326d-8390-46b4-aa47-95c3beca6550, so in the topology .m4 file
+tools/topology/m4/src.m4:
+
+.. code-block:: none
+
+   dnl Defines the macro for SRC widget
+   define(SRC_UUID_ARRAY, UUID_ARRAY("c1c5326d-8390-46b4-aa47-95c3beca6550"))
+
+We have a .m4 macro UUID_ARRAY() will help to transfer format in #2 (36
+digits) to 16 Bytes byte array, after compiled, the corresponding ALSA
+UUID token defined here will be generated:
+
+.. code-block:: none
+
+   /* vendor tuple for uuid */
+   struct snd_soc_tplg_vendor_uuid_elem {
+           __le32 token;
+           char uuid[16];
+   } __attribute__((packed));
+
+Linux topology driver
+*********************
+The topology driver will parse the 16Bytes UUID token and sent it to the
+firmware in component creation stage, **for all components**.
+
+.. code-block:: none
+
+   /* Generic component tokens */
+   static const struct sof_topology_token comp_tokens[] = {
+           {SOF_TKN_COMP_CORE_ID,
+               SND_SOC_TPLG_TUPLE_TYPE_WORD, get_token_u32,
+               offsetof(struct sof_ipc_comp, core), 0},
+           {SOF_TKN_COMP_UUID,
+               SND_SOC_TPLG_TUPLE_TYPE_UUID, get_token_uuid,
+               offsetof(struct sof_ipc_comp, uuid), 0},
+   };
+
+.. code-block:: none
+
+   static int sof_widget_ready(struct snd_soc_component *scomp, int index,
+                               struct snd_soc_dapm_widget *w,
+                               struct snd_soc_tplg_dapm_widget *tw)
+   {
+   ...
+           ret = sof_parse_tokens(scomp, &template, comp_tokens,
+                                  ARRAY_SIZE(comp_tokens), tw->priv.array,
+                                  le32_to_cpu(tw->priv.size));
+   ...
+   }
+
+UUID arrays stored section
+**************************
+The UUID arrays are stored to SRAM as a separated section, though it
```

Comment:
A general documentation should avoid references to specific memory types since they may be completely different for various platforms. And the whole concept of the dedicated section is that it is outside the run-time memory space even (SRAM on some systems).

Reply:
Thanks. Since we have decided to split and save only part of those UUIDs to rodata section, I would need to refine this part anyway.

Reply:
the PR is updated with this part changed, mark it 'resolved'.